core/src/main/java/org/apache/calcite/rel/type/DelegatingTypeSystem.java

```diff
@@ -52,6 +52,14 @@ protected DelegatingTypeSystem(RelDataTypeSystem typeSystem) {
     return typeSystem.getMaxNumericPrecision();
   }
 
+  @Override public boolean isSupportNegativeScale() {
```

Comment:
I think you should open a new issue for this change.

Reply:
And probably you don't need two verbs, you can call it supportsNegativeScale.

Reply:
This PR only includes some tests so we can close the Jira issue.